scipy/stats/tests/test_distributions.py

```diff
@@ -7569,3 +7569,73 @@ def test_moment_order_4():
     # has since been made more accurate and efficient, so now this test
     # is expected to pass.
     assert stats.skewnorm._munp(4, 0) == 3.0
+
+
+class TestRelativisticBW:
+    @pytest.mark.parametrize(
+        "rho,gamma", [
+            (36.545206797050334, 2.4952),  # Z0 Boson
+            (38.55107913669065, 2.085),  # W Boson,
+        ]
+    )
+    def test_pdf_integration(self, rho, gamma):
+        """Test pdf integrates to 1."""
+        res = quad(stats.relativistic_bw.pdf, 0, np.inf, args=(rho, 0, gamma))
+        assert_allclose(res[0], 1)
```

Comment:
I don't think this one is in [`test_continuous_basic`](https://github.com/scipy/scipy/blob/main/scipy/stats/tests/test_continuous_basic.py), but personally I wouldn't include tests that are applicable to all distributions in distribution-specific tests.

Reply:
Just a note from 10000 feet that there is a generic test of this kind: https://github.com/scipy/scipy/blob/main/scipy/stats/tests/common_tests.py#L23

Reply:
Great. So I think all of these are implemented generically.

Reply:
I see. I had looked at the PR for the most recently distribution added distribution (studentized range) and added the same tests that appeared there.  I noticed there was some overlap with the generic tests but ended up thinking it was good to be able to test with sets of parameters that correspond to actual particles.  The Higgs boson test identified issues with fitting I wouldn’t have found otherwise for instance but couldn’t be used in the generic tests because as far as I can tell, distr_params won’t allow using a scale parameter. 

Also, are there generic tests for fitting with floc set? This seems to be the only case where fitting rbw will actually work. 

I’ll leave it up to you two whether we should take these tests out though.

Reply:
Tests from `common_tests.py` take values from https://github.com/scipy/scipy/blob/main/scipy/stats/_distr_params.py --- and these latter values end up in the default docstring examples. So all in all it sounds like the actual Higgs mass (or is it $Z^0$ boson) could/should very well end up in this list.

Whether to keep tests or not --- I'm not current on recent enhancements in stats (as is, over last several years), so am not making decisions here :-).

Reply:
Maybe the best solution would be to allow distr_params to take loc and scale parameters. I’m not sure how much work this would take though.

Reply:
ISTM this would be reasonably easy (and it would be great indeed): 
the loop over the distributions,  https://github.com/scipy/scipy/blob/main/scipy/stats/tests/test_continuous_basic.py#L146 , takes `args`
 https://github.com/scipy/scipy/blob/main/scipy/stats/tests/test_continuous_basic.py#L142
which come straight from the list in `_distr_params.py`. So I *think* simply adding values for loc and scale to the relevant line in https://github.com/scipy/scipy/blob/main/scipy/stats/_distr_params.py#L6 should just work.

Whether the docstring generation needs a tweak for the case where loc and scale are given, I don't know, and suspect only that the only way is to step through https://github.com/scipy/scipy/blob/main/scipy/stats/_distn_infrastructure.py#L755
Most likely it'll need a slight tweak of the generic doctsting example template.

Reply:
Ok cool. I was getting an error when I tried that but will look into again when time permits. It seems like the error should be an easy fix

Reply:
Oops. There are some things about studentized range I'd change if I were starting it now!

You can include any of these if they're not redundant. They just looked redundant, or if they weren't, I was suggesting that they be added generically.

It's definitely fine to keep the fit tests or improve the generic ones. It would be good to run the `fit` test with user-defined `scale` because code paths for those are often parameter or data dependent.

Reply:
Ok, I think the best thing to do is to keep only the fit tests here and figure out why I was getting an error when trying to add loc and scale to distr_params, then use the params for the Higgs boson there.

Reply:
> So I think simply adding values for loc and scale to the relevant line in https://github.com/scipy/scipy/blob/main/scipy/stats/_distr_params.py#L6 should just work.

Nope, it doesn't _just work_ :-). The reason is several tests either assume the default loc and scale (hence asserts fail), or supply `loc=0, scale=1` explicitly (hence multiple values for a loc keyword), or assume a signature of `*args, n, loc=0, scale=1` (e.g. `moment`).

So to allow custom loc and scale would need to augment the list in `_distr_param` with a dict of loc and scale keywords, , where applicable, make individual `check_` routines receive this extra `kwds` dict as a parameter (also handle the case where it is not supplied) and use pass them to `distfn` methods. 

While this is not too difficult per se, it definitely does not block this PR.

Reply:
Thanks for looking into that @ev-br! @mdhaber do you think it’s ok to keep some of these tests (maybe just for the most difficult case of the Higgs) until we make these changes in distr_params?

Reply:
> The Higgs boson test identified issues with fitting I wouldn’t have found otherwise

I see that adding `loc`/`scale` to the `fit` test would be useful, and I agree that we should be able to add `loc` and `scale` to `distr_params` (so that all these parameters are shown in the documentation, which would be nice here). I don't really see why `loc` and `scale` are needed for the other tests, since those parameters are handled in a generic way. Note that you can add multiple sets of parameters to `distr_params` if that's what you're going for - only one will be rendered. 

But if you prefer to include them here, I suppose they're simple enough. Please just explain as a comment in the code for posterity.

Reply:
> I'm happy to use custom `loc` and `scale` in the other tests, but I'm not sure why they would be needed, since those parameters are handled in a generic way.

I thought it would be good to test with actual parameter values for real particles. In that case scale will be equal to the decay width. It would have been embarrassing if some numerical issue caused problems for one of the  combinations that will actually be used. That’s what the fit tests identified for the Higgs Boson, which led to overriding fit. 

I think I’m going to get rid of all of the dist specific tests except the fitting ones though. Later I can follow up adding more combinations to distr_params after it’s able to take loc/scale parameters.